### DISCUSSION TITLE
Is there a way to rewind the Ink story back to the last choice? Equivalent to the "Back" button in the Inky editor?

### DISCUSSION THREAD

**Discussion by Nerrolken** [inkle/ink#677] [Game integration help]
My integration is using Ink to drive a series of random event popups, with the text of the popup coming from __inkStory.Continue()_ and the choices being buttons in the popup.

My problem is, if the player ignores the popup and chooses nothing (for example quitting the app with the popup still onscreen), the next time the popup appears there's no text, because that __inkStory.Continue()_ is still stuck at the choices.

A perfect solution would be something like the "Back" button in the Inky editor, returning you to the last choice point so you can continue again from there. Does something like that exist in the runtime API?

**Comment by toolness** [comment]
Hello! I'm not sure if anything like this exists in the runtime API, but I was thinking that the same kind of thing would be useful, and one thought I had was to [save the state of the game](https://github.com/inkle/ink/blob/master/Documentation/RunningYourInk.md#saving-and-loading) on every interaction, and restore it if the user decided to use the "back" button. Does that help?
